code/modules/station_goals/dna_vault.dm

```diff
@@ -0,0 +1,190 @@
+//Crew has to create dna vault
+// Cargo can order DNA samplers + DNA vault boards
+// DNA vault requires x animals ,y plants, z human dna
+// DNA vaults require high tier stock parts and cold
+/datum/station_goal/dna_vault
+	name = "DNA Vault"
+	var/animal_count
+	var/human_count
+	var/plant_count
+
+/datum/station_goal/dna_vault/New()
+	..()
+	animal_count = rand(15,20) //might be too few given ~15 roundstart stationside ones
+	human_count = rand(round(0.75 * ticker.totalPlayersReady) , ticker.totalPlayersReady) // 75%+ roundstart population.
+	var/non_standard_plants = non_standard_plants_count()
+	plant_count = rand(round(0.7 * non_standard_plants),round(0.9 * non_standard_plants))
+
+/datum/station_goal/dna_vault/proc/non_standard_plants_count()
+	. = 0
+	for(var/T in subtypesof(/obj/item/seeds)) //put a cache if it's used anywhere else
+		var/obj/item/seeds/S = T
+		if(initial(S.rarity) > 0)
+			.++
+
+/datum/station_goal/dna_vault/get_report()
+	return {"Our long term prediction systems say there's 99% chance of system-wide cataclysm in near future.
+	 We need you to construct DNA Vault aboard your station.
+
+	 DNA Vault needs to contain samples of:
+	 [animal_count] unique animal data
+	 [plant_count] unique non-standard plant data
+	 [human_count] unique sapient humanoid DNA data
+
+	 Base vault parts should be availible for shipping by your cargo shuttle."}
+
+
+/datum/station_goal/dna_vault/on_report()
+	var/datum/supply_pack/P = SSshuttle.supply_packs[/datum/supply_pack/misc/dna_vault]
+	P.special_enabled = TRUE
+
+	P = SSshuttle.supply_packs[/datum/supply_pack/misc/dna_probes]
+	P.special_enabled = TRUE
+
+/datum/station_goal/dna_vault/check_completion()
+	for(var/obj/machinery/dna_vault/V in machines)
+		if(V.animals.len >= animal_count && V.plants.len >= plant_count && V.dna >= human_count)
+			return TRUE
+	return FALSE
+
+
+/obj/item/device/dna_probe
+	name = "DNA Sampler"
+	desc = "Can be used to take chemical and genetic samples of pretty much anything."
+	icon = 'icons/obj/syringe.dmi'
+	item_state = "hypo"
+	icon_state = "hypo"
+	flags = NOBLUDGEON
+	var/list/animals = list()
+	var/list/plants = list()
+	var/list/dna = list()
+
+/obj/item/device/dna_probe/proc/clear_data()
+	animals = list()
+	plants = list()
+	dna = list()
+
+var/list/non_simple_animals = typecacheof(list(/mob/living/carbon/monkey)|subtypesof(/mob/living/carbon/alien))
```

Comment:
this is a weird line, why not ```typecacheof(list(/mob/living/carbon/monkey, /mob/living/carbon/alien))```
(yes I know atm you're ignoring the spooky alien base path, but nothing uses that anyway, bar real aliens)